spec/features/comment_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.feature 'Book commentary', type: :feature do
+  subject do
+    visit book_path(random_book)
+
+    fill_in 'You can live your comment here', with: comment
+    click_button 'Add comment'
+  end
+
+  let(:user)        { create(:user) }
+  let(:comment)     { Faker::JapaneseMedia::OnePiece.quote }
+  let(:book_list)   { create_list(:book, 4) }
+  let(:random_book) { book_list.sample }
+
+  before do
+    book_list
+    Book.reindex
+  end
+
+  context 'when user doesnt authenticate' do
```

Comment:
'when user is not logged in'

hmm, it seems that you redirect unauthenticated user when he tries to leave the comment. It make sense not to show comment form for unauthenticated users at all